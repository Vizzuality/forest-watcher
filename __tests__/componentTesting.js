import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import ActionCard from '../app/components/common/action-card';

// Small component ui testing doesn't look to be very useful,
// but this means we should be able to test screen component render method outputs.
describe('Component Tests', () => {
  it('test runs', () => {
    expect(true);
  });

  it('ActionCard renders', () => {
    const tree = renderer.create(
      <ActionCard label="test_label" />
    );
    expect(tree).toMatchSnapshot();
  });

  it('ActionCard renders all props', () => {
    const tree = renderer.create(
      <ActionCard label="test_label" width={10} height={20} icon={1234} action={() => {}} />
    );
    expect(tree).toMatchSnapshot();
  });
});
