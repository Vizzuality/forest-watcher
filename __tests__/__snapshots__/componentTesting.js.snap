// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Tests ActionCard renders 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#97be32",
          "borderStyle": "dashed",
          "borderWidth": 2,
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
      ]
    }
  >
    <Image
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#97be32",
        }
      }
    >
      test_label
    </Text>
  </View>
</View>
`;

exports[`Component Tests ActionCard renders all props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#97be32",
          "borderStyle": "dashed",
          "borderWidth": 2,
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "height": 20,
          "width": 10,
        },
      ]
    }
  >
    <Image
      source={1234}
      style={
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "resizeMode": "contain",
          "width": 24,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#97be32",
        }
      }
    >
      test_label
    </Text>
  </View>
</View>
`;
